### ISSUE TITLE
Unicode issue in API Name

### ISSUE THREAD

**Issue by dnplus** [Axway-API-Management-Plus/apim-cli#113]
**API-Mgr. CLI version**
> Please make sure to use the latest available version before reporting a bug.

1.3.0

**API-Manager and Service-Pack Version**
7.7.20200130

**Expected behavior**
The API name should be support Unicode 

**Actual behavior**
OAuth整合API => OAuth??API, but only the name is broken, seems the content is fine.

**Log-Output in Debug**
<img width="782" alt="CleanShot 2020-11-11 at 10 03 11@2x" src="https://user-images.githubusercontent.com/2977735/98756452-8d5c3980-2405-11eb-9ff4-32c0c9f5e836.png">

**Test case sample**
> Please describe the steps you have done to replicate the issue

Export API that API name contains Unicode character, all export config in UTF8 and no garbled.
Then import it.

**Comment by cwiechmann** [comment]
Thanks for reporting this issue @dnplus. There is already an integration test in place, that validates proper Unicode-Character handling based on the API-Summary field. Both, the API-Summary and Name is configured in the same API-Config file and obviously treated in the same way.
Unfortunately, I cannot include the name-field as part of that test, as it's used to locate the imported API during the test. However, I did a quick test by changing the name and run the test. The imported API looks as expected:
![image](https://user-images.githubusercontent.com/16292002/98781551-ce991d00-23f6-11eb-978d-667afe16b75a.png)

The API-Config file I used is UTF-8 encoded and this is expected by the CLI. I also tested to export the API and Re-Imported it and it works as well. 
Can you please: 
- verify the encoding of your configuration file?
- test with the API-Summary field?

**Comment by dnplus** [comment]
Seems encoding is right.
![CleanShot 2020-11-12 at 11 43 39@2x](https://user-images.githubusercontent.com/2977735/98893114-c0bdc780-24dc-11eb-85b1-efcb0637859b.png)

Same result.

**Comment by cwiechmann** [comment]
Hmm, yes that looks perfectly fine! And is the summary information shown okay in the UI or is it only the API-Name?

**Comment by dnplus** [comment]
Not sure the summary should appear in FE or BE, according to another issue the import stopped in BE.

Seems the BE didn't apply this summary information, but all swagger content is fine, only the API-Name has problem.

**Comment by cwiechmann** [comment]
Yes, the summary information appears in the FE-API. 
The content of the Swagger-File is not so important, as it's handled differently then the API-Config file. It is more or less send straight to the API-Manager. 

The screenshot you provided in the other issue is helpful as it shows, that the API-Name is incorrect, but the API-Description looks good. 
![image](https://user-images.githubusercontent.com/16292002/98909687-73346100-24c2-11eb-8b67-64c980d150f0.png)
Both, API-Name and API-Description, are coming from the config-file and normally all fields from the config-file are handled in the same way. But obviously somewhere between reading the config with UTF-8 encoding and the final POST/PUT-Request the API-Name got corrupted.

My Unicode Integration-Test is importing a fresh new API (not updating an existing). Can you confirm that the issue also happens when importing the API as a new one?

**Comment by dnplus** [comment]
Just change the basepath and name, it's the same.
![CleanShot 2020-11-12 at 16 08 34@2x](https://user-images.githubusercontent.com/2977735/98912665-5ec38900-2501-11eb-8d12-22d4ca3c5bd1.png)

app-config
```
{
  "name" : "OAuth整合API-TEST2",
  "apiSummary": "許功蓋",
  "path" : "/api/oauth/integrate/test2",
  "state" : "unpublished",
  "version" : "1.0",
  "organization" : "API Development",
  "apiDefinition" : "OAuth整合API.yaml",
  "backendBasepath" : "http://localhost:8080",
  "tags" : {
    "GSS" : [ "TWOB" ]
  },
  "caCerts" : [ {
    "certFile" : "SamplesTestCA.crt",
    "inbound" : "true",
    "outbound" : "false"
  } ],
  "clientOrganizations" : [ "API Development" ],
  "applications" : [ {
    "name" : "絲綢記帳",
    "enabled" : true,
    "state" : "approved",
    "oauthResources" : [ ]
  } ]
}
```

result:
```
----------------------------------------------------------------------------------------
API-Manager CLI: 1.3.0

To report issues or get help, please visit:
https://github.com/Axway-API-Management-Plus/apim-cli
----------------------------------------------------------------------------------------
Module: API - I M P O R T (1.3.0)
----------------------------------------------------------------------------------------
750    INFO     APIManagerAdapter| Successfully connected to API-Manager (7.7.20200130) on: https://api-env:8075
866    INFO  IImportConfigAdapter| Reading API-Definition (Swagger/WSDL) from file: '/Users/dnplus/Downloads/apim-cli-1.3.0/scripts/api-oauth-integrate/OAuth整合API.yaml' (absolute path)
982    INFO  IImportConfigAdapter| Handling configured client-applications.
1039   INFO  IImportConfigAdapter| Found existing application: '絲綢記帳' (null) based on given name '絲綢記帳'
1208   INFO      APIImportManager| No existing API found, creating new!
1460   ERROR         APIImportApp| Can't import definition / Create BE-API.
com.axway.apim.lib.errorHandling.AppException: Can't import definition / Create BE-API.
	at com.axway.apim.adapter.apis.APIManagerAPIAdapter.importBackendAPI(APIManagerAPIAdapter.java:726)
	at com.axway.apim.apiimport.actions.CreateNewAPI.execute(CreateNewAPI.java:40)
	at com.axway.apim.apiimport.APIImportManager.applyChanges(APIImportManager.java:50)
	at com.axway.apim.APIImportApp.importAPI(APIImportApp.java:103)
	at com.axway.apim.APIImportApp.importAPI(APIImportApp.java:59)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.axway.apim.cli.APIManagerCLI.run(APIManagerCLI.java:142)
	at com.axway.apim.cli.APIManagerCLI.main(APIManagerCLI.java:78)
Caused by: java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Long.parseLong(Long.java:601)
	at java.lang.Long.parseLong(Long.java:631)
	at com.axway.apim.adapter.apis.APIManagerAPIAdapter.importBackendAPI(APIManagerAPIAdapter.java:723)
	... 10 more
```

**Comment by cwiechmann** [comment]
I realized, that the Backend-API-Name is wrong and not the Frontend-API. I should have read your description better.

I made a small change that fixes the problem:
![image](https://user-images.githubusercontent.com/16292002/98915436-32404a80-24ca-11eb-8d7e-4c81a7709e87.png)

As before, you can download this snapshot release that includes the fixes for issue #112 and #113. I'm going to delete the other build to avoid confusion:
[axway-apimcli-1.3.1-SNAPSHOT.zip](https://github.com/Axway-API-Management-Plus/apim-cli/files/5529234/axway-apimcli-1.3.1-SNAPSHOT.zip)
